csrc/preseg_passes/remove_bcast_squeeze.cpp

```diff
@@ -389,6 +390,16 @@ TensorView* maybeDoReplacement(TensorView* orig) {
     }
   }
 
+  // If we are replacing an output, we need to preserve the memory layout by
```

Comment:
Can we do this check in `Fusion::replaceOutput`?

Reply:
That check needs to be opt-in. We have places where we are explicitly altering memory layout. i.e. mark alias prepare and sharding propagation that would intentionally change the memory layout.

But it would be nice to have a check on that.

So the check would actually be checking for compatible transforms. i.e. the two transforms might not necessarily be the same, but they should be equivalent... This feels like something @wujingyue and @zasdfgbnm were discussing together.

Reply:
Opened this to track #3994